docs/running-on-kubernetes.md

```diff
@@ -51,87 +51,15 @@ connect without SSL on a different port, the master would be set to `k8s://http:
-      --master k8s://192.168.99.100 \
-      --conf spark.kubernetes.driver.uploads.jars=/home/exampleuser/exampleapplication/dep1.jar,/home/exampleuser/exampleapplication/dep2.jar \
-      --conf spark.kubernetes.driver.docker.image=registry-host:5000/spark-driver:latest \
-      --conf spark.kubernetes.executor.docker.image=registry-host:5000/spark-executor:latest \
-      /home/exampleuser/exampleapplication/main.jar
-
-To specify a main application resource that can be downloaded from an HTTP service, and if a plugin for that application
-is located in the jar `/opt/spark-plugins/app-plugin.jar` on the docker image's disk:
-
-    bin/spark-submit \
-      --deploy-mode cluster \
-      --class com.example.applications.PluggableApplication \
-      --master k8s://192.168.99.100 \
-      --jars /opt/spark-plugins/app-plugin.jar \
-      --conf spark.kubernetes.driver.docker.image=registry-host:5000/spark-driver-custom:latest \
-      --conf spark.kubernetes.executor.docker.image=registry-host:5000/spark-executor:latest \
-      http://example.com:8080/applications/sparkpluggable/app.jar
-      
-Note that since passing the jars through the `--jars` command line argument is equivalent to setting the `spark.jars`
-Spark property, the above will behave identically to this command:
-
-    bin/spark-submit \
-      --deploy-mode cluster \
-      --class com.example.applications.PluggableApplication \
-      --master k8s://192.168.99.100 \
-      --conf spark.jars=file:///opt/spark-plugins/app-plugin.jar \
-      --conf spark.kubernetes.driver.docker.image=registry-host:5000/spark-driver-custom:latest \
-      --conf spark.kubernetes.executor.docker.image=registry-host:5000/spark-executor:latest \
-      http://example.com:8080/applications/sparkpluggable/app.jar
-      
-To specify a main application resource that is in the Docker image, and if it has no other dependencies:
-
-    bin/spark-submit \
-      --deploy-mode cluster \
-      --class com.example.applications.PluggableApplication \
-      --master k8s://192.168.99.100:8443 \
-      --conf spark.kubernetes.driver.docker.image=registry-host:5000/spark-driver-custom:latest \
-      --conf spark.kubernetes.executor.docker.image=registry-host:5000/spark-executor:latest \
-      container:///home/applications/examples/example.jar
+### Dependency Management and Docker Containers
 
+Spark supports specifying JAR paths that are either on the submitting host's disk, or are located on the disk of the
+driver and executors. Refer to the [application submission](submitting-applications.html#advanced-dependency-management)
+section for details. Note that files specified with the `local` scheme should be added to the container image of both
+the driver and the executors. Files without a scheme or with the scheme `file://` are treated as being on the disk of
+the submitting machine, and are uploaded to the driver running in Kubernetes before launching the application.
+ 
```

Comment:
The examples would merely be re-hashing what is already stated in the advanced dependency management system. The additional library jars on executor containers are just files specified with `local://` and those files also have to be on the driver container... which is also covered in the advanced dependency management section. Similarly with added local files.

The SSL configuration is interesting however - I don't think the docs have been updated to reflect that.